drivers/flash/flash_stm32u5x.c

```diff
@@ -0,0 +1,380 @@
+/*
+ * Copyright (c) 2022 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_DOMAIN flash_stm32u5
+#define LOG_LEVEL CONFIG_FLASH_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_DOMAIN);
+
+#include <kernel.h>
+#include <device.h>
+#include <string.h>
+#include <drivers/flash.h>
+#include <init.h>
+#include <soc.h>
+#include <stm32u5xx_ll_icache.h>
+#include <stm32_ll_system.h>
+
+#include "flash_stm32.h"
+
+/* at this time stm32u5 mcus have 1MB (stm32u575) or 2MB (stm32u585) */
+#define STM32U5_SERIES_MAX_FLASH	2048
+#define BANK2_OFFSET	(KB(STM32U5_SERIES_MAX_FLASH) / 2)
+
+#define ICACHE_DISABLE_TIMEOUT_VALUE           1U   /* 1ms */
+#define ICACHE_INVALIDATE_TIMEOUT_VALUE        1U   /* 1ms */
+
+static int stm32u5x_icache_disable(void)
```

Comment:
I wonder what is the diff vs L5 driver.
If close enough, it would be good to use a single file to ease maintainance.
For instance https://github.com/zephyrproject-rtos/zephyr/pull/42099 may need to be backported on this file also is duplicated